### DISCUSSION TITLE
Milvus Build issue

### DISCUSSION THREAD

**Discussion by chetan-shah01** [milvus-io/milvus#37165] [Q&A and General discussion]
Hi,

While Building Milvus using source code version 2.4.13, getting below error :-

 -- Getting CMake variables for x64-linux
     -- Getting CMake variables for x64-linux-dbg
     -- Getting CMake variables for x64-linux-rel
     -- Configuring x64-linux-dbg
     CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
         Command failed: /usr/bin/bash -c "V=1 ./../src/nssl-3.3.0-961d78b8a1.clean/vcpkg/configure  \"/usr/bin/perl\" \"/workspace/cmake_build/azure/_deps/vcpkg-src/buildtrees/openssl/src/nssl-3.3.0-961d78b8a1.clean/Configure\" \"linux-x86_64\" \"enable-static-engine\" \"enable-capieng\" \"no-tests\" \"no-docs\" \"no-shared\" \"no-module\" \"no-apps\" \"--openssldir=/etc/ssl\" \"--libdir=lib\" \"--disable-silent-rules\" \"--verbose\" \"--disable-shared\" \"--enable-static\" \"--debug\" \"--prefix=/workspace/cmake_build/azure/vcpkg_installed/x64-linux/debug\""
         Working Directory: /workspace/cmake_build/azure/_deps/vcpkg-src/buildtrees/openssl/x64-linux-dbg
         Error code: 2
         See logs for more information:
           /workspace/cmake_build/azure/_deps/vcpkg-src/buildtrees/openssl/config-x64-linux-dbg-err.log
     
     Call Stack (most recent call first):
       scripts/cmake/vcpkg_configure_make.cmake:863 (vcpkg_execute_required_process)
       ports/openssl/unix/portfile.cmake:106 (vcpkg_configure_make)
       ports/openssl/portfile.cmake:79 (include)
       scripts/ports.cmake:175 (include)
     
     
     error: building openssl:x64-linux failed with: BUILD_FAILED
     Elapsed time to handle openssl:x64-linux: 3.7 s
     Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
     Then check for known issues at:
       https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+openssl
     You can submit a new issue at:
       https://github.com/microsoft/vcpkg/issues/new?title=[openssl]+Build+error+on+x64-linux&body=Copy+issue+body+from+%2Fworkspace%2Fcmake_build%2Fazure%2Fvcpkg_installed%2Fvcpkg%2Fissue_body.md
     
     -- Running vcpkg install - failed
     CMake Error at /workspace/cmake_build/azure/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:899 (message):
       vcpkg install failed.  See logs for more information:
       /workspace/cmake_build/azure/vcpkg-manifest-install.log
     Call Stack (most recent call first):
       /usr/local/share/cmake-3.26/Modules/CMakeDetermineSystem.cmake:148 (include)
       CMakeLists.txt:22 (project)

Can you please suggest on this ?

**Comment by xiaofan-luan** [comment]
what is your linux version?

try to use -Z so to skip Azure build see if this can be solved

**Comment by yhmo** [comment]
Add a line in the scripts/core_build.sh and try again:
`BUILD_WITHOUT_AZURE="ON"`

**Reply by chetan-shah01** [reply]
Thanks @yhmo  Above issue get fixed after adding "BUILD_WITHOUT_AZURE="ON" and build move forward.
Now getting cmake error "
CMake Error at CMakeLists.txt:99 (include):
       include could not find requested file:
         /workspace/cmake_build/conan/conanbuildinfo.cmake
     CMake Error at CMakeLists.txt:101 (conan_basic_setup):
       Unknown CMake command "conan_basic_setup".

As checked on google, 
https://stackoverflow.com/questions/60967172/how-to-fix-unknown-command-error-in-cmake-when-i-using-conan, need to add cmake generators in file "cmake_build/conan/conanbuildinfo.cmake" but still issue is persisting so can you suggest on this ?

**Reply by xiaofan-luan** [reply]
did you do ./scripts/install_deps.sh before you do make?

seems that you didn't have conan installed

**Comment by chetan-shah01** [comment]
Hi @xiaofan-luan,  Yes I am installing the dependencies before running make .

Please find below reference logs:

   Running setup.py install for pluginbase ... - done
         Running setup.py install for patch-ng ... - done
       WARNING: The script distro is installed in '/usr/local/bin' which is not on PATH.
       Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
         Running setup.py install for conan ... - \ | / done
     Successfully installed Jinja2-3.1.4 MarkupSafe-3.0.2 PyJWT-2.9.0 PyYAML-6.0.2 bottle-0.12.25 colorama-0.4.6 conan-1.64.1 distro-1.8.0 fasteners-0.19 node-semver-0.6.1 patch-ng-1.17.4 pluginbase-1.0.1 pygments-2.18.0 tqdm-4.66.6 urllib3-1.26.20

**Reply by chetan-shah01** [reply]
Hi @xiaofan-luan,

Getting CMake error while building milvus , can you suggest on this ? 

CMake Error at /usr/local/share/cmake-3.26/Modules/CMakeTestCXXCompiler.cmake:60 (message):
       The C++ compiler
     
         "/usr/bin/c++"
     
       is not able to compile a simple test program.
     
       It fails with the following output:
     
         Change Dir: /root/.conan/data/ninja/1.11.1/_/_/build/24647d9fe8ec489125dfbae4b3ebefaf7581674c/build/Release/CMakeFiles/CMakeScratch/TryCompile-Xua3nb

**Comment by xiaofan-luan** [comment]
try to export PATH=$PATH:/usr/local/bin

and check conan --version 

what did you you get? let's check if conan installed successfully .

If we get all the log of your compile it would be great

**Comment by chetan-shah01** [comment]
Hi @xiaofan-luan ,

Conan version 1.64.1 and where I add "export PATH=$PATH:/usr/local/bin"

and below is link for compile logs

[compile logs.txt](https://github.com/user-attachments/files/17780148/compile.logs.txt)

**Comment by xiaofan-luan** [comment]
/usr/bin/c++ -Wno-error=address -Wno-error=deprecated-declarations   -m64  -static-libstdc++ -static-libgcc  CMakeFiles/cmTC_dfdeb.dir/testCXXCompiler.cxx.o -o cmTC_dfdeb 
         /usr/bin/ld: cannot find -lstdc++

this is the core error. try to check if conan is working with right cpp version, here is what shows on my laptop

![Uploading image.png…]()

**Comment by xiaofan-luan** [comment]
[settings]
os=Macos
os_build=Macos
arch=x86_64
arch_build=x86_64
compiler=clang
compiler.version=15
compiler.libcxx=libc++
build_type=Release
[options]
[conf]
[build_requires]
[env]

**Comment by chetan-shah01** [comment]
Hi @xiaofan-luan , Can you suggest where I add "export PATH=$PATH:/usr/local/bin" to fix this /usr/bin/ld: cannot find -lstdc++

**Reply by xiaofan-luan** [reply]
you will need to export this in your bash profile and and check your conan config at ~/.conan/profiles

**Comment by chetan-shah01** [comment]
Hi @xiaofan-luan, I am building the milvus using RIO pipeline. 

I can see libstdc++ in below path and I have made the symlink for this lib so that c++ compiler can find libstdc++ to compile the program but it is not working. 

+ find /usr -name 'libstdc++.so*'
     /usr/lib/gcc/x86_64-redhat-linux/11/32/libstdc++.so
     /usr/lib/gcc/x86_64-redhat-linux/11/libstdc++.so
     /usr/lib64/libstdc++.so.6
     /usr/lib64/libstdc++.so.6.0.29
     /usr/share/gdb/auto-load/usr/lib64/__pycache__/libstdc++.so.6.0.29-gdb.cpython-39.opt-1.pyc
     /usr/share/gdb/auto-load/usr/lib64/__pycache__/libstdc++.so.6.0.29-gdb.cpython-39.pyc
     /usr/share/gdb/auto-load/usr/lib64/libstdc++.so.6.0.29-gdb.py

     + ln -s /usr/lib64/libstdc++.so.6 /usr/lib/libstdc++.so

     + /usr/bin/ld --verbose
     + tr -s ' ;' '\012'
     + grep SEARCH_DIR
     SEARCH_DIR("=/usr/x86_64-redhat-linux/lib64")
     SEARCH_DIR("=/usr/lib64")
     SEARCH_DIR("=/usr/local/lib64")
     SEARCH_DIR("=/lib64")
     SEARCH_DIR("=/usr/x86_64-redhat-linux/lib")
     SEARCH_DIR("=/usr/local/lib")
     SEARCH_DIR("=/lib")
     SEARCH_DIR("=/usr/lib")

**Comment by xiaofan-luan** [comment]
not an expert on compilation. maybe @Presburger can help?

**Comment by chetan-shah01** [comment]
Hi @Presburger, while building milvus , getting below error

Can you suggest on this ?

CMake Error at /usr/local/share/cmake-3.28/Modules/CMakeTestCXXCompiler.cmake:60 (message):
       The C++ compiler
     
         "/usr/bin/c++"
     
       is not able to compile a simple test program.
     
       It fails with the following output:
     
         Change Dir: '/root/.conan/data/ninja/1.11.1/_/_/build/24647d9fe8ec489125dfbae4b3ebefaf7581674c/build/Release/CMakeFiles/CMakeScratch/TryCompile-aQ3QyE'
         
         Run Build Command(s): /usr/local/bin/cmake -E env VERBOSE=1 /usr/bin/gmake -f Makefile cmTC_8fc5c/fast
         gmake[1]: Entering directory '/root/.conan/data/ninja/1.11.1/_/_/build/24647d9fe8ec489125dfbae4b3ebefaf7581674c/build/Release/CMakeFiles/CMakeScratch/TryCompile-aQ3QyE'
         /usr/bin/gmake  -f CMakeFiles/cmTC_8fc5c.dir/build.make CMakeFiles/cmTC_8fc5c.dir/build
         gmake[2]: Entering directory '/root/.conan/data/ninja/1.11.1/_/_/build/24647d9fe8ec489125dfbae4b3ebefaf7581674c/build/Release/CMakeFiles/CMakeScratch/TryCompile-aQ3QyE'
         Building CXX object CMakeFiles/cmTC_8fc5c.dir/testCXXCompiler.cxx.o
         /usr/bin/c++   -Wno-error=address -Wno-error=deprecated-declarations   -m64  -o CMakeFiles/cmTC_8fc5c.dir/testCXXCompiler.cxx.o -c /root/.conan/data/ninja/1.11.1/_/_/build/24647d9fe8ec489125dfbae4b3ebefaf7581674c/build/Release/CMakeFiles/CMakeScratch/TryCompile-aQ3QyE/testCXXCompiler.cxx
         Linking CXX executable cmTC_8fc5c
         /usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_8fc5c.dir/link.txt --verbose=1
         /usr/bin/c++ -Wno-error=address -Wno-error=deprecated-declarations   -m64  -static-libstdc++ -static-libgcc  CMakeFiles/cmTC_8fc5c.dir/testCXXCompiler.cxx.o -o cmTC_8fc5c 
         /usr/bin/ld: cannot find -lstdc++
         collect2: error: ld returned 1 exit status
         gmake[2]: *** [CMakeFiles/cmTC_8fc5c.dir/build.make:99: cmTC_8fc5c] Error 1
         gmake[2]: Leaving directory '/root/.conan/data/ninja/1.11.1/_/_/build/24647d9fe8ec489125dfbae4b3ebefaf7581674c/build/Release/CMakeFiles/CMakeScratch/TryCompile-aQ3QyE'
         gmake[1]: *** [Makefile:127: cmTC_8fc5c/fast] Error 2
         gmake[1]: Leaving directory '/root/.conan/data/ninja/1.11.1/_/_/build/24647d9fe8ec489125dfbae4b3ebefaf7581674c/build/Release/CMakeFiles/CMakeScratch/TryCompile-aQ3QyE
     
       CMake will not be able to correctly generate this project.
     Call Stack (most recent call first):
       CMakeLists.txt:6 (project)
     
     -- Configuring incomplete, errors occurred!
     ninja/1.11.1: 
     ninja/1.11.1: ERROR: Package '24647d9fe8ec489125dfbae4b3ebefaf7581674c' build failed
     ninja/1.11.1: WARN: Build folder /root/.conan/data/ninja/1.11.1/_/_/build/24647d9fe8ec489125dfbae4b3ebefaf7581674c/build/Release
     ERROR: ninja/1.11.1: Error in build() method, line 40
     	cmake.configure()
     	ConanException: Error 1 while executing cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/root/.conan/data/ninja/1.11.1/_/_/build/24647d9fe8ec489125dfbae4b3ebefaf7581674c/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/root/.conan/data/ninja/1.11.1/_/_/package/24647d9fe8ec489125dfbae4b3ebefaf7581674c" -DCMAKE_EXE_LINKER_FLAGS="-static-libstdc++ -static-libgcc" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/root/.conan/data/ninja/1.11.1/_/_/build/24647d9fe8ec489125dfbae4b3ebefaf7581674c/src"
     conan install failed
